proxy/envoy/egress.go

```diff
@@ -42,20 +50,26 @@ func buildEgressRoutes(services model.ServiceDiscovery, mesh *proxyconfig.ProxyM
 }
 
 // buildEgressRoute translates an egress rule to an Envoy route
-func buildEgressHTTPRoute(svc *model.Service) *VirtualHost {
+func buildEgressHTTPRoute(mesh *proxyconfig.ProxyMeshConfig, svc *model.Service) *VirtualHost {
 	var host *VirtualHost
 
 	for _, servicePort := range svc.Ports {
 		protocol := servicePort.Protocol
 		switch protocol {
 		case model.ProtocolHTTP, model.ProtocolHTTP2, model.ProtocolGRPC, model.ProtocolHTTPS:
-			cluster := buildOutboundCluster(svc.Hostname, servicePort, nil)
+			key := svc.Key(servicePort, nil)
+			cluster := &Cluster{
+				Name:   fmt.Sprintf("%x", sha1.Sum([]byte(key))),
+				Type:   ClusterTypeStrictDNS,
+				LbType: DefaultLbType,
+				Hosts: []Host{{
+					URL: fmt.Sprintf("tcp://%s:%d", svc.ExternalName, servicePort.Port),
+				}},
+			}
 
-			// overwrite cluster hosts and types
-			cluster.Type = ClusterTypeStrictDNS
-			cluster.Hosts = []Host{{
-				URL: fmt.Sprintf("tcp://%s:%d", svc.ExternalName, servicePort.Port),
-			}}
+			if servicePort.Protocol == model.ProtocolGRPC || servicePort.Protocol == model.ProtocolHTTP2 {
+				cluster.Features = ClusterFeatureHTTP2
+			}
```

Comment:
If auth is enabled, I believe we need to set alpn_protocol or something to http2. Not sure if that is handled elsewhere